src/Microsoft.DotNet.Build.Tasks/BinClashLogger.cs

```diff
@@ -0,0 +1,231 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class BinClashLogger : ILogger
+    {
+        internal class ProjectState
+        {
+            public ProjectState(ProjectStartedEventArgs args)
+            {
+                GlobalProperties = args.GlobalProperties;
```

Comment:
It looks to me like its a detached copy of the properties, created with ToDictionary().  Also, this is a logger, it's purpose is to observe.  If it was possible to change the collection that'd be a bug in MSBuild and not something we should rely on.